docs/linker-attributes.md

```diff
@@ -0,0 +1,78 @@
+# Linker attributes
+
+As the linker scans the referenced code, it warns about non-understood reflection patterns or other code patterns whose behavior may not be preserved by linking. The following attributes influence the pattern-detection and warning behavior.
```

Comment:
I think this should be a general intro about where the attributes are and how to use them. It'd not pick reflection only summary here at all